src/azure-cli/azure/cli/command_modules/servicefabric/_params.py

```diff
@@ -111,3 +113,108 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
+        c.argument('consider_warning_as_error', options_list=['--warning-as-error', '--consider-warning-as-error'], arg_type=get_three_state_flag(),
+                   help='Indicates whether to treat a warning health event as an error event during health evaluation.')
+        c.argument('default_service_type_max_percent_unhealthy_partitions_per_service', options_list=['--max-porcent-unhealthy-partitions'],
+                   help='Specify the maximum percent of unhelthy partitions per service allowed by the health policy for the default service type to use for the monitored upgrade. Allowed values are form 0 to 100.')
+        c.argument('default_service_type_max_percent_unhealthy_replicas_per_partition', options_list=['--max-porcent-unhealthy-replicas'],
+                   help='Specify the maximum percent of unhelthy replicas per service allowed by the health policy for the default service type to use for the monitored upgrade. Allowed values are form 0 to 100.')
+        c.argument('default_service_type_max_percent_unhealthy_services', options_list=['--max-porcent-unhealthy-services'],
+                   help='Specify the maximum percent of unhelthy services allowed by the health policy for the default service type to use for the monitored upgrade. Allowed values are form 0 to 100.')
+        c.argument('max_percent_unhealthy_deployed_applications', options_list=['--max-porcent-unhealthy-apps'],
+                   help='Specify the mximum percentage of the application instances deployed on the nodes in the cluster that have a health state of error before the application health state for the cluster is error. Allowed values are form 0 to 100.')
+
+    with self.argument_context('sf application create', validator=validate_create_application) as c:
+        c.argument('application_type_name', options_list=['--type-name', '--application-type-name'], help='Specify the application type name.')
+        c.argument('application_type_version', options_list=['--version', '--application-type-version'], help='Specify the application type version.')
+        c.argument('package_url', help='Specify the url of the application package sfpkg file.')
+        c.argument('application_parameters', options_list=['--parameters', '--application-parameters'], action=addAppParamsAction, nargs='+',
+                   help='Specify the application parameters as key/value pairs. These parameters must exist in the application manifest.'
+                   'for example: --application-parameters param1=value1 param2=value2')
+        c.argument('minimum_nodes', options_list=['--min-nodes', '--minimum-nodes'],
+                   help='Specify the minimum number of nodes where Service Fabric will reserve capacity for this application.')
+        c.argument('maximum_nodes', options_list=['--max-nodes', '--maximum-nodes'],
+                   help='Specify the maximum number of nodes on which to place an application.')
```

Comment:
It looks there are many parameters you are using many times. You can use CLIArgumentType to abstract them and reuse them.

Reply:
ok thanks for the suggestion I have register the common parameters at the top